keras/backend/tensorflow_backend.py

```diff
@@ -193,6 +201,64 @@ def set_session(session):
     _SESSION = session
 
 
+# DEVICE MANIPULATION AND PROBING
+
+class _TfDeviceCaptureOp(object):
+    """Class for capturing the tf device scope."""
+
+    def __init__(self):
+        """__init__ method."""
+        self.device = None
+
+    def _set_device(self, device):
+        """_set_device method."""
+        self.device = device
+
+
+def get_current_tf_device():
```

Comment:
I think we should make this function private, as well as `is_current_explicit_device`, and `get_available_gpus`.